local/compose/create.go

```diff
@@ -602,32 +644,48 @@ func getAliases(s types.ServiceConfig, c *types.ServiceNetworkConfig) []string {
 	return aliases
 }
 
-func getNetworkMode(p *types.Project, service types.ServiceConfig) container.NetworkMode {
+func getNetworkMode(ctx context.Context, c client.APIClient, p *types.Project, service types.ServiceConfig) (container.NetworkMode, error) {
 	mode := service.NetworkMode
 	if mode == "" {
 		if len(p.Networks) > 0 {
 			for name := range getNetworksForService(service) {
-				return container.NetworkMode(p.Networks[name].Name)
+				return container.NetworkMode(p.Networks[name].Name), nil
 			}
 		}
-		return container.NetworkMode("none")
+		return container.NetworkMode("none"), nil
 	}
 
-	// FIXME incomplete implementation
+	serviceName := ""
 	if strings.HasPrefix(mode, "service:") {
-		panic("Not yet implemented")
-	}
-	if strings.HasPrefix(mode, "container:") {
-		panic("Not yet implemented")
+		_, _ = fmt.Sscanf(mode, "service:%s", &serviceName)
+		serviceContainers, err := c.ContainerList(ctx, moby.ContainerListOptions{
```

Comment:
better use s.getContainers which also handle oneOff flag and return a `Containers` type we can extend for a (bit more) functional design

Reply:
By this do you mean to write a method `GetCotainers` in `ServiceConfig` because I couldn't find such method

Reply:
https://github.com/docker/compose-cli/blob/9063c138ba13d154d03d3da5b399141ea6bb944f/local/compose/containers.go#L40

Reply:
in terms of architecture, it seems to me this should be fully avoided as the caller is `createContainer` in convergence.go which already knows the "actual state" of services containers. 

Maybe we should make `convergence` a dedicated struct, which owns `actualState` and create/recreate functions